anemo/src/crypto.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+static SUPPORTED_SIG_ALGS: &[&webpki::SignatureAlgorithm] = &[&webpki::ED25519];
+
+#[derive(Clone, Debug)]
+pub(crate) struct CertVerifier(pub(crate) String);
```

Comment:
You could directly use https://github.com/MystenLabs/mysten-infra/blob/97844c8bb9307d518e879969da527911b5ab9034/rccheck/src/lib.rs#L167-L288 which is just built on a simple generic trait:
https://github.com/MystenLabs/mysten-infra/blob/97844c8bb9307d518e879969da527911b5ab9034/rccheck/src/lib.rs#L45-L58